File: web/src/lib/severity.ts
// Canonical policy severity levels, most-severe first. Mirrors the backend
// models.PolicySeverities and the DB check constraint — keep in sync.

export const severityOptions: { value: string; label: string }[] = [
  { value: 'critical', label: 'Critical' },
  { value: 'high', label: 'High' },
  { value: 'medium', label: 'Medium' },
  { value: 'low', label: 'Low' },
  { value: 'info', label: 'Info' }
]

// Rank for sorting, derived from the canonical order (critical = 0 … info = 4).
export const severityRank: Record<string, number> = Object.fromEntries(
  severityOptions.map((o, i) => [o.value, i])
)

// Maps a severity to its badge variant (oat data-variant).
export const severityVariant: Record<string, string> = {
  critical: 'danger',
  high: 'danger',
  medium: 'warning',
  low: 'info',
  info: 'info'
}

File: internal/models/severity.go
package models

import "strings"

// PolicySeverities is the canonical, ordered (most-severe first) list of policy
// severity levels. It is the single source of truth for the Go runtime; the DB
// CHECK constraint, the request-validation `oneof` tags, and the frontend
// severity options must mirror it.
var PolicySeverities = []string{"critical", "high", "medium", "low", "info"}

// DefaultPolicySeverity is used when a severity is empty or unrecognized. It
// matches the DB column default.
const DefaultPolicySeverity = "medium"

var policySeveritySet = func() map[string]bool {
	m := make(map[string]bool, len(PolicySeverities))
	for _, s := range PolicySeverities {
		m[s] = true
	}
	return m
}()

// IsValidPolicySeverity reports whether s is a recognized severity level.
func IsValidPolicySeverity(s string) bool {
	return policySeveritySet[s]
}

// NormalizePolicySeverity lowercases/trims the severity and falls back to
// DefaultPolicySeverity for empty or unrecognized values.
func NormalizePolicySeverity(s string) string {
	s = strings.ToLower(strings.TrimSpace(s))
	if policySeveritySet[s] {
		return s
	}
	return DefaultPolicySeverity
}
